test/mocha/10-api.js

```diff
@@ -59,4 +59,41 @@ describe('bedrock-config-yaml', () => {
       testEnvValue: 123123123123
     });
   });
+  it('does not expose data when a load from env error throws', async () => {
+    // Include a bad config with duplicate key
+    const yaml = `
+    app:
+      sensitive: 1337
+      sensitive: hello-world
+    core:
+      test-core-env: 987553
+    `;
+
+    const encodedConfig = Buffer.from(yaml).toString('base64');
+    process.env.BEDROCK_CONFIG = encodedConfig;
+
+    let output = '';
+    await events.emit('bedrock.configure').catch(e => {
+      output = e.message;
+    }).finally(() => {
+      delete process.env.BEDROCK_CONFIG;
+    });
+
+    expect(output).to.not.include('1337');
```

Comment:
Does this assertion pass too ... for extra safety?

```suggestion
    expect(output).to.not.include('1337');
    expect(output).to.not.include('hello-world');
```